src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+(define-read-only (cost_err_cons (n uint))
+    (runtime u219))
+
+(define-read-only (cost_default_to (n uint))
+    (runtime u249))
+
+(define-read-only (cost_unwrap_ret (n uint))
+    (runtime u299))
+
+(define-read-only (cost_unwrap_err_or_ret (n uint))
+    (runtime u339))
+
+(define-read-only (cost_is_okay (n uint))
+    (runtime u287))
+
+(define-read-only (cost_is_none (n uint))
+    (runtime u212))
+
+(define-read-only (cost_is_err (n uint))
+    (runtime u282))
+
+(define-read-only (cost_is_some (n uint))
+    (runtime u223))
+
+(define-read-only (cost_unwrap (n uint))
+    (runtime u284))
+
+(define-read-only (cost_unwrap_err (n uint))
+    (runtime u264))
+
+(define-read-only (cost_try_ret (n uint))
+    (runtime u256))
+
+(define-read-only (cost_match (n uint))
+    (runtime u286))
+
+(define-read-only (cost_or (n uint))
+    (runtime (linear n u3 u149)))
+
+(define-read-only (cost_and (n uint))
+    (runtime (linear n u3 u149)))
+
+(define-read-only (cost_append (n uint))
+    (runtime (linear n u71 u176)))
+
+(define-read-only (cost_concat (n uint))
+    (runtime (linear n u75 u244)))
+
+(define-read-only (cost_as_max_len (n uint))
+    (runtime u475))
+
+(define-read-only (cost_contract_call (n uint))
+    (runtime u153))
+
+(define-read-only (cost_contract_of (n uint))
+    (runtime u13400))
+
+(define-read-only (cost_principal_of (n uint))
+    (runtime u39))
+
+
+(define-read-only (cost_at_block (n uint))
+    {
+        runtime: u210,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_load_contract (n uint))
+    {
+        runtime: (linear n u1 u157),
+        write_length: u0,
+        write_count: u0,
+        ;; set to 3 because of the associated metadata loads
+        read_count: u3,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_create_map (n uint))
+    {
+        runtime: (linear n u1 u1631),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_var (n uint))
+    {
+        runtime: (linear n u7 u2152),
+        write_length: (linear n u1 u1),
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_nft (n uint))
+    {
+        runtime: (linear n u1 u1610),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_ft (n uint))
+    {
+        runtime: u1972,
+        write_length: u1,
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_fetch_entry (n uint))
+    {
+        runtime: (linear n u1 u1539),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_entry (n uint))
```

Comment:
This is the same for deletions, right?

Reply:
Yep -- though deletion costs will almost always be cheaper, because the input is just the key size, not the value size as well.